### PR TITLE
Some fixes / tweaks in Execution namepace

### PR DESCRIPTION
**Pull request by lucasmichot**
- [ ] Added or updated tests
- [ ] Added Docs for all relevant versions

**Related Issue/Intent**

- Fix some missing imports
- Split long code into many methods
- promote early return
- some clean-up on duplicate code
- some docblocks fixes

### CODE REVIEW

File: src/Execution/DataLoader/BatchLoader.php [hunk 1]
```diff
@@ -65,7 +66,7 @@ public static function instance(string $loaderClass, array $pathToField, array $
             );
 
         if (! $instance instanceof self) {
-            throw new \Exception(
+            throw new Exception(
```
Comment (lucasmichot) [inline src/Execution/DataLoader/BatchLoader.php]: Import.

File: src/Execution/DataLoader/ModelRelationFetcher.php [hunk 2]
```diff
@@ -257,22 +257,21 @@ function (Model $model) use ($relationName, $relationConstraints) {
...
-            $withProperty = $reflection->getProperty('with');
-            $withProperty->setAccessible(true);
-
-            $with = array_filter(
-                (array) $withProperty->getValue($model),
-                function ($relation) use ($model) {
-                    return ! $model->relationLoaded($relation);
-                }
-            );
+        if ($collection->isEmpty()) {
```
Comment (lucasmichot) [inline src/Execution/DataLoader/ModelRelationFetcher.php]: Prefer early return.

File: src/Execution/ErrorBuffer.php [hunk 3]
```diff
@@ -85,12 +85,10 @@ public function push(string $errorMessage, ?string $key = null): self
     {
         if ($key === null) {
             $this->errors[] = $errorMessage;
-
-            return $this;
+        } else {
```
Comment (lucasmichot) [inline src/Execution/ErrorBuffer.php]: No need to duplicate return code.

File: src/Execution/ErrorHandler.php [hunk 4]
```diff
@@ -16,5 +17,5 @@ interface ErrorHandler
      * @param  \Closure  $next
      * @return array
      */
-    public static function handle(Error $error, \Closure $next);
+    public static function handle(Error $error, Closure $next): array;
```
Comment (lucasmichot) [inline src/Execution/ErrorHandler.php]: Missing import and type hint `array`

File: src/Execution/ExtensionErrorHandler.php [hunk 5]
```diff
@@ -16,7 +17,7 @@ class ExtensionErrorHandler implements ErrorHandler
      * @param  \Closure  $next
      * @return array
      */
-    public static function handle(Error $error, \Closure $next): array
+    public static function handle(Error $error, Closure $next): array
```
Comment (lucasmichot) [inline src/Execution/ExtensionErrorHandler.php]: Missing import.

File: src/Execution/MutationExecutor.php [hunk 6]
```diff
@@ -2,6 +2,8 @@
 
 namespace Nuwave\Lighthouse\Execution;
 
+use ReflectionClass;
```
Comment (lucasmichot) [inline src/Execution/MutationExecutor.php]: Missing imports

File: src/Execution/MutationExecutor.php [hunk 7]
```diff
@@ -42,88 +46,32 @@ public static function executeCreate(Model $model, Collection $args, Relation $p
...
-            /** @var \Illuminate\Database\Eloquent\Relations\MorphOne $relation */
-            $relation = $model->{$relationName}();
-
-            collect($nestedOperations)->each(function ($values, string $operationKey) use ($relation): void {
-                if ($operationKey === 'create') {
-                    self::handleSingleRelationCreate(collect($values), $relation);
-                }
-            });
-        });
+        self::executeCreateHasMany($model, $hasMany);
```
Comment (lucasmichot) [inline src/Execution/MutationExecutor.php]: Split this code into dedicated methods.

File: src/Execution/QueryFilter.php [hunk 8]
```diff
@@ -48,9 +48,9 @@ public static function getInstance(FieldValue $value): self
             .'.'.strtolower($value->getFieldName());
 
         // Get an existing instance or register a new one
-        return app()->has($handler)
-            ? resolve($handler)
-            : app()->instance($handler, resolve(static::class));
+        return app()->bound($handler)
```
Comment (lucasmichot) [inline src/Execution/QueryFilter.php]: prefer `bound` and `app`: `has` and `resolve` are just aliases

File: src/Execution/Utils/Cursor.php [hunk 9]
```diff
@@ -27,11 +27,11 @@ class Cursor
      */
     public static function decode(array $args): int
     {
-        if (! $cursor = Arr::get($args, 'after')) {
-            return 0;
+        if ($cursor = Arr::get($args, 'after')) {
```
Comment (lucasmichot) [inline src/Execution/Utils/Cursor.php]: prefer early return.

File: src/Execution/Utils/Pagination.php [hunk 10]
```diff
@@ -15,7 +15,7 @@ class Pagination
     public static function calculateCurrentPage(int $first, int $after, int $defaultPage = 1): int
     {
         return $first && $after
-            ? floor(($first + $after) / $first)
+            ? (int) (($first + $after) / $first)
```
Comment (lucasmichot) [inline src/Execution/Utils/Pagination.php]: enfore return type to int

File: src/Execution/Utils/Subscription.php [hunk 11]
```diff
@@ -2,6 +2,8 @@
 
 namespace Nuwave\Lighthouse\Execution\Utils;
 
+use Throwable;
```
Comment (lucasmichot) [inline src/Execution/Utils/Subscription.php]: missing imports

**Comment by spawnia** [comment]
Thanks for the work @lucasmichot, much appreciated.

**Pull request merged**